### ISSUE TITLE
PyTorch training loop

### ISSUE THREAD

**Issue by isaacmg** [AIStream-Peelout/flow-forecast#17]
- [x] Write PyTorch training loop compatible with any base time series PyTorch model 
- [x] Initialize opt, criterion, epochs, learning rate, etc based on params 
- [x] Log epochs to `self.params`. Write Tensorboard and Wandb data if applicable based on the params.
- [x] Unit tests. Specifically, test with MultiHeadAttn. Use single epoch and part of data.
